cmd/searcher/search/search_structural.go

```diff
@@ -289,7 +289,11 @@ func structuralSearchWithZoekt(ctx context.Context, p *protocol.Request) (matche
 			Languages:                    p.Languages,
 		}
 
-	repoBranches := map[string][]string{string(p.Repo): {string(p.Branch)}}
+	branch := p.Branch
+	if branch == "" {
+		branch = "HEAD"
+	}
+	repoBranches := map[string][]string{string(p.Repo): {branch}}
```

Comment:
```suggestion
	if p.Branch == "" {
		p.Branch = "HEAD"
	}
	repoBranches := map[string][]string{string(p.Repo): {p.Branch}}
```

I think this is fine?

Reply:
Seems fine to me. Doesn't look like `p` is used further down, so mutating it shouldn't hurt anything